.github/workflows/ci.yml

```diff
@@ -63,3 +66,100 @@ jobs:
     - name: Build kube-router
       run: |
         make kube-router
+
+  # Builds Container on Push Event, not run if there's a Tag Event
+  ci-build-container-push:
+    runs-on: ubuntu-latest
+    if: ${{ startsWith('refs/tags/v', github.ref) != true && github.event_name != 'pull_request'  }}
+    steps:
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v1
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+
+    - name: Login to DockerHub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+    - name: Build and push
+      uses: docker/build-push-action@v2
+      with:
+        context: .
+        platforms: linux/amd64,linux/arm64,linux/arm,linux/ppc64,linux/s390x
+        push: true
+        tags: cloudnativelabs/kube-router:${{ github.sha }}
+
+  # Builds Container on Push Event, not run if there's a Tag Event
+  ci-build-container-pr:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request'
+    steps:
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v1
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+
+    - name: Login to DockerHub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
```

Comment:
Do GitHub actions give you an easy way to re-use this logic, since it appears to be mostly the same in all steps where we want to build and push a container?

Reply:
I was trying to get it right first and then optimize. I changed it to DRY and keep my fingers crossed.